zerver/views/webhooks/wordpress.py

```diff
@@ -0,0 +1,52 @@
+# Webhooks for external integrations.
+from __future__ import absolute_import
+from django.utils.translation import ugettext as _
+from django.http import HttpRequest, HttpResponse
+from zerver.models import Client, get_client, UserProfile
+from zerver.lib.actions import check_send_message
+from zerver.lib.response import json_success, json_error
+from zerver.decorator import REQ, has_request_variables, api_key_only_webhook_view
+
+from six import text_type
+
+@api_key_only_webhook_view("Wordpress")
+@has_request_variables
+def api_wordpress_webhook(request, user_profile, client,
+                          stream=REQ(default="wordpress"),
+                          topic=REQ(default="WordPress Notification"),
+                          hook=REQ(default="WordPress Action"),
+                          post_title=REQ(default="New WordPress Post"),
+                          post_type=REQ(default="post"),
+                          post_url=REQ(default="WordPress Post URL"),
+                          display_name=REQ(default="New User Name"),
+                          user_email=REQ(default="New User Email"),
+                          user_login=REQ(default="Logged in User")):
+    # type: (HttpRequest, UserProfile, Client, text_type, text_type, text_type, text_type, text_type, text_type, text_type, text_type, text_type) -> HttpResponse
+
+    # remove trailing whitespace (issue for some test fixtures)
+    hook = hook.rstrip()
+    post_title = post_title.rstrip()
```

Comment:
Which values need to be stripped depends on if they come last in the test fixture. For hook, I have a test that includes only that, so I can't rearrange the parameters in the test fixture to avoid the problem. (Which I did for another one.) I can rearrange some additional fixture data if you'd prefer to only strip ones minimally necessary. 

The current code is the result of trying to manage long lines of code constructing the message bodies.